src/pages/index.js

```diff
@@ -1,45 +1,354 @@
-import appConfig from 'appConfig';
 import Head from 'next/head';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import polyline from '@mapbox/polyline';
+import { useSelector } from 'react-redux';
+
+import Controls from 'components/Controls';
+import Directions from 'components/Directions';
+import Elevation from 'components/Elevation';
+import Map from 'components/Map';
+import TitleBar from 'components/Titlebar';
+import WelcomeModal from 'components/WelcomeModal';
+
+import { getRoute } from 'lib/api';
+import { logQuery } from 'lib/analytics';
+import { handleError } from 'lib/error';
+import { geocode, reverseGeocode } from 'lib/geocode';
+import { latlngIsWithinBounds, updateMapSize, getPathDistance } from 'lib/map';
+import { updateUrlParams, readUrlParams, validateUrlParams } from 'lib/url';
+
+import {
+  clearPath,
+  setStartLocation,
+  setStartAddress,
+  setEndLocation,
+  setEndAddress,
+  setPath,
+  setDistance,
+  setDirections,
+  setElevationProfile,
+} from '@redux/slices/search';
```

Comment:
Importing individual actions from Redux instead of setting into state. Some of these will likely get cleaned up or grouped together in subsequent passes.